### ISSUE TITLE
Error: Building package openssl-windows:x86-windows failed with: BUILD_FAILED

### ISSUE THREAD

**Issue by jonjow** [microsoft/vcpkg#10647]
**Host Environment**
- OS: Windows 10

**To Reproduce**
`./vcpkg install openssl`

**Failure logs** 
PS C:\svn\vcpkg> .\vcpkg install openssl
Computing installation plan...
A suitable version of cmake was not found (required v3.14.0). Downloading portable cmake v3.14.0...
Downloading cmake...
  https://github.com/Kitware/CMake/releases/download/v3.14.0/cmake-3.14.0-win32-x86.zip -> C:\svn\vcpkg\downloads\cmake-3.14.0-win32-x86.zip
Extracting cmake...
A suitable version of 7zip was not found (required v18.1.0). Downloading portable 7zip v18.1.0...
Downloading 7zip...
  https://www.nuget.org/api/v2/package/7-Zip.CommandLine/18.1.0 -> C:\svn\vcpkg\downloads\7-zip.commandline.18.1.0.nupkg
Extracting 7zip...
A suitable version of nuget was not found (required v4.6.2). Downloading portable nuget v4.6.2...
Downloading nuget...
  https://dist.nuget.org/win-x86-commandline/v4.6.2/nuget.exe -> C:\svn\vcpkg\downloads\nuget.exe
The following packages will be built and installed:
    openssl[core]:x86-windows
  * openssl-windows[core]:x86-windows
Additional packages (*) will be modified to complete this operation.
A suitable version of powershell-core was not found (required v6.2.1). Downloading portable powershell-core v6.2.1...
Downloading powershell-core...
  https://github.com/PowerShell/PowerShell/releases/download/v6.2.1/PowerShell-6.2.1-win-x86.zip -> C:\svn\vcpkg\downloads\PowerShell-6.2.1-win-x86.zip
Extracting powershell-core...
Starting package 1/2: openssl-windows:x86-windows
Building package openssl-windows[core]:x86-windows...
-- Downloading http://strawberryperl.com/download/5.30.0.1/strawberry-perl-5.30.0.1-32bit.zip...
-- Downloading https://www.openssl.org/source/openssl-1.1.1d.tar.gz...
-- Downloading https://www.openssl.org/source/openssl-1.1.1d.tar.gz... Failed. Status: 22;"HTTP response code said error"
-- Downloading https://www.openssl.org/source/old/1.1.1/openssl-1.1.1d.tar.gz...
-- Extracting source C:/svn/vcpkg/downloads/openssl-1.1.1d.tar.gz
-- Using source at C:/svn/vcpkg/buildtrees/openssl-windows/src/openssl-1-b1f8ae3e6c
-- Downloading http://www.nasm.us/pub/nasm/releasebuilds/2.14.02/win32/nasm-2.14.02-win32.zip...
-- Downloading http://download.qt.io/official_releases/jom/jom_1_1_3.zip...
-- Downloading http://download.qt.io/official_releases/jom/jom_1_1_3.zip... Failed. Status: 56;"Failure when receiving data from the peer"
-- Downloading http://mirrors.ocf.berkeley.edu/qt/official_releases/jom/jom_1_1_3.zip...
-- Downloading http://mirrors.ocf.berkeley.edu/qt/official_releases/jom/jom_1_1_3.zip... Failed. Status: 56;"Failure when receiving data from the peer"
CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:175 (message):

      Failed to download file.
      If you use a proxy, please set the HTTPS_PROXY and HTTP_PROXY environment
      variables to "https://user:password@your-proxy-ip-address:port/".
      Otherwise, please submit an issue at https://github.com/Microsoft/vcpkg/issues

Call Stack (most recent call first):
  scripts/cmake/vcpkg_find_acquire_program.cmake:343 (vcpkg_download_distfile)
  ports/openssl-windows/portfile.cmake:31 (vcpkg_find_acquire_program)
  scripts/ports.cmake:90 (include)

Error: Building package openssl-windows:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: openssl-windows:x86-windows
  Vcpkg version: 2020.02.04-nohash

Additionally, attach any relevant sections from the log files above.

**Additional context**
PS C:\svn\vcpkg> perl -v

This is perl 5, version 30, subversion 1 (v5.30.1) built for MSWin32-x64-multi-thread

Copyright 1987-2019, Larry Wall

Perl may be copied only under the terms of either the Artistic License or the
GNU General Public License, which may be found in the Perl 5 source kit.

Complete documentation for Perl, including FAQ lists, should be found on
this system using "man perl" or "perldoc perl".  If you have access to the
Internet, point your browser at http://www.perl.org/, the Perl Home Page.

**Comment by heydojo** [comment]
@jonjow looks like your net connection might be being throttled. Either by ISP via traffic shaping, a data limit or maybe even by a router QOS rule.

If you try `./vcpkg install openssl` again after the download failures, (provided the throttling of your connection subsides,) the files will likely download and the build complete.
The file archives which have already been downloaded are cached locally. And as such won't be downloaded again, therefore allowing bandwidth to be used to try to download the required files which failed.

My opinion here is that vcpkg should maybe retry a bit more aggressively and offer a warning that the net connection may not be supplying the required bandwidth to complete the download or some other such, in cases like the one you have reported here.

In short: I've seen this message. It was a bandwidth problem. I re-ran the same vcpkg command again after the failure, the files then did download and the operation completed successfully. Please try that.

**Comment by Voskrese** [comment]
> -- Downloading http://mirrors.ocf.berkeley.edu/qt/official_releases/jom/jom_1_1_3.zip... Failed. Status: 56;"Failure when receiving data from the peer"

**Comment by linquize** [comment]
Manually put the zip files and put in `vcpkg/downloads` directory

**Comment by jonjow** [comment]
I fixed the problem. It was because my project was running from a network drive...
